Use d8 on CI

kripken opened the pull request:
Adds the v8 shell to the list of JS engines so the test runner can use it.

Replace spidermonkey in `other` tests with v8, which will let those tests be run now. (The reason spidermonkey was there is that a long time ago it was the simplest way to get a JS VM with wasm in it, before wasm showed up in stable releases.)

A bunch of tiny test fixes that this caught, like

 * Show `Enlarging memory arrays, this is not fast` warning on wasm2js, because like in asm.js, growth there means a big copy of the entire memory.
 * In wasm synchronous compilation mode, handle not just the spidermonkey error text for a bad memory size but also the v8 text.
 * Remove a monotonic timer test that was v8 specific and has not been run for many years - it seems like the old test depended on the d8 timer being monotonic, and it seems like that's changed (maybe it behaves more like `Date.now()`), as this isn't a stable interface we just shouldn't test it.

sbc100 commented on tests/test_other.py:
```diff
@@ -6118,12 +6118,12 @@ def test_massive_alloc(self):
     run_process([PYTHON, EMCC, 'main.cpp', '-s', 'ALLOW_MEMORY_GROWTH=1'])
     # no message about growth, just check return code
-    run_js('a.out.js', stderr=PIPE, full_output=True, engine=SPIDERMONKEY_ENGINE)
+    run_js('a.out.js', stderr=PIPE, full_output=True, engine=V8_ENGINE)
```
Can node still not do these things?

kripken replied:
Good point, yeah, node can do them these days. So we could just remove all these special cases. That would be simpler, for a small loss of more testing on a single shell. Maybe that's better though, what do you think?

sbc100 replied:
I don't think we can really consider d8 and node to be different engines since they are based on the same codebase.    I'm tempted to limit emscripten to node LTS + browsers only to keep things simple in the testing matrix.

kripken commented:
@sbc100 Makes sense, I agree.

Ok, I updated the PR to not use spidermonkey *or* v8 in those tests, and just use normal node. This actually found another bug! The warning about memory size didn't throw the original exception, which ended up hanging node.js. Fixed.

Overall, this PR's description is now:

 * Add v8 to CI, but don't run any tests with it yet. This may be useful soon for testing new wasm features not yet in stable node, though.
 * Simplify testing code to use node instead of spidermonkey or v8, where we used those because we assumed node didn't have wasm yet. Fix various small bugs that follow.
 * Show Enlarging memory arrays, this is not fast warning on wasm2js, because like in asm.js, growth there means a big copy of the entire memory.
 * In wasm synchronous compilation mode, handle not just the spidermonkey error text for a bad memory size but also the v8 text. Also throw the error properly.
 * Remove a monotonic timer test that was v8 specific and has not been run for many years - it seems like the old test depended on the d8 timer being monotonic, and it seems like that's changed (maybe it behaves more like Date.now()), as this isn't a stable interface we just shouldn't test it.

sbc100 commented on .circleci/config.yml:
```diff
@@ -342,6 +342,8 @@ jobs:
             # we are trying to test.
             rm -Rf `find -name "emscripten"`
+            echo "V8_ENGINE = '`pwd`/upstream/bin/d8'" >> ~/.emscripten
+            echo "JS_ENGINES = [NODE_JS, V8_ENGINE]" >> ~/.emscripten
```
Can this be reverted now?

kripken replied:
I think it's still useful for two things: we do test it on some `other` tests (which iterate on JS_ENGINES) which makes sure we remain compatible with the `d8` shell. Also I just realized while talking to @tlively that we *do* have simd tests that use d8 if present, and this pr did run them and they passed :)

tlively replied:
Once SIMD in node stabilizes I should probably update the simd tests to run in node as well.
